File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # частота
        freq = {}
        for val in nums:
            freq[val] = freq.get(val, 0) + 1
        
        buckets = []
        for i in range(len(nums)+1):
            buckets.append([])

        for num, count in freq.items():
            buckets[count].append(num)

        ans = []
        for i in range(len(buckets)-1, 0, -1):
            for num in buckets[i]:
                ans.append(num)
                if len(ans) == k:
                    return ans
        return ans

        

File: Data Structures & Algorithms/remove-element/submission-3.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k =0;
        int r = 0; 
        for (int l=0; l<nums.length; l++){
            if (val != nums[l]){
                int temp = nums[l];
                nums[l] = nums[r];
                nums[r] = temp;
                k++;
                r++;
                
            }

        }
        return k; 
    }
}